packages/dd-trace/test/llmobs/sdk/index.spec.js

```diff
@@ -0,0 +1,1008 @@
+            span = tracer.scope().active()
+            sinon.spy(span, 'finish')
+          })
+
+          fn()
+
+          expect(span.finish).to.have.been.called
+        })
+
+        it('wraps a function with a callback', () => {
+          let span
+          let next
+
+          const fn = llmobs.wrap({ kind: 'workflow' }, (_next) => {
+            span = tracer.scope().active()
+            sinon.spy(span, 'finish')
+            next = _next
+          })
+
+          fn(() => {})
+
+          expect(span.finish).to.not.have.been.called
+
+          next()
+
+          expect(span.finish).to.have.been.called
+        })
+
+        it('does not auto-annotate llm spans', () => {
+          let span
+          function myLLM (input) {
+            span = llmobs._active()
+            return ''
+          }
+
+          const wrappedMyLLM = llmobs.wrap({ kind: 'llm' }, myLLM)
+
+          wrappedMyLLM('input')
+
+          expect(LLMObsTagger.tagMap.get(span)).to.deep.equal({
+            '_ml_obs.meta.span.kind': 'llm',
+            '_ml_obs.meta.ml_app': 'mlApp',
+            '_ml_obs.llmobs_parent_id': 'undefined'
+          })
+        })
+
+        it('does not auto-annotate embedding spans input', () => {
+          let span
+          function myEmbedding (input) {
+            span = llmobs._active()
+            return 'output'
+          }
+
+          const wrappedMyEmbedding = llmobs.wrap({ kind: 'embedding' }, myEmbedding)
+
+          wrappedMyEmbedding('input')
+
+          expect(LLMObsTagger.tagMap.get(span)).to.deep.equal({
+            '_ml_obs.meta.span.kind': 'embedding',
+            '_ml_obs.meta.ml_app': 'mlApp',
+            '_ml_obs.llmobs_parent_id': 'undefined',
+            '_ml_obs.meta.output.value': 'output'
+          })
+        })
+
+        it('does not auto-annotate retrieval spans output', () => {
+          let span
+          function myRetrieval (input) {
+            span = llmobs._active()
+            return 'output'
+          }
+
+          const wrappedMyRetrieval = llmobs.wrap({ kind: 'retrieval' }, myRetrieval)
+
+          wrappedMyRetrieval('input')
+
+          expect(LLMObsTagger.tagMap.get(span)).to.deep.equal({
+            '_ml_obs.meta.span.kind': 'retrieval',
+            '_ml_obs.meta.ml_app': 'mlApp',
+            '_ml_obs.llmobs_parent_id': 'undefined',
+            '_ml_obs.meta.input.value': 'input'
+          })
+        })
+
+        // need span kind optional for this test
+        // it('sets the span name to "unnamed-anonymous-function" if no name is provided', () => {
+        //   let span
+        //   const fn = llmobs.wrap({ kind: 'workflow' }, () => {
+        //     span = llmobs._active()
+        //   })
+
+        //   fn()
+
+        //   expect(span.context()._name).to.equal('unnamed-anonymous-function')
+        // })
+      })
+
+      describe('parentage', () => {
+        // it('starts a span with a distinct trace id', () => {
```

Comment:
A lot of tests seem to be commented out without an explanation. It would be better to `it.skip` them and add an explanation.